classes/Kohana/Jam/Field/Price.php

```diff
@@ -48,9 +48,9 @@ protected function _default(Jam_Validated $model, $value)
 		$return = FALSE;
 
 		$value = $this->run_filters($model, $value);
-        if (is_string($value)) {
-            $value = (float) $value;
-        }
+		if ((is_string($value) AND is_numeric($value)) OR is_int($value) ) {
+			$value = (float) $value;
+		}
 
 		// Convert empty values to NULL, if needed
 		if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0)
```

Comment:
@EZarkov, so after the `is_int` check and convert to float we can remove this check `AND $value !== 0`, right?